pkg/v5/configmap/desired.go

```diff
@@ -11,29 +11,45 @@ import (
 	"github.com/giantswarm/cluster-operator/pkg/label"
 )
 
+type configMapGenerator func(ctx context.Context, configMapValues ConfigMapValues, projectName string) (*corev1.ConfigMap, error)
+
+type BasicConfigMap struct {
+	Image Image `json:"image"`
+}
+
+type IngressController struct {
+	Controller IngressControllerController `json:"controller"`
+}
+
+type IngressControllerController struct {
+	Replicas int   `json:"replicas"`
+	Image    Image `json:"image"`
+}
+
+type Image struct {
+	Registry string `json:"registry"`
+}
+
 func (s *Service) GetDesiredState(ctx context.Context, configMapValues ConfigMapValues) ([]*corev1.ConfigMap, error) {
 	desiredConfigMaps := make([]*corev1.ConfigMap, 0)
 
-	{
-		configMap, err := s.newIngressControllerConfigMap(ctx, configMapValues, s.projectName)
+	generators := []configMapGenerator{
+		s.newIngressControllerConfigMap,
+		s.newKubeStateMetricsConfigMap,
+		s.newNodeExporterConfigMap,
```

Comment:
I like the generator concept being introduced here. <3

Reply:
Thanks! :) Here the intention is only not repeating code, but yes, maybe the concept can be useful in general.